src/dmd/root/filename.d

```diff
@@ -997,25 +997,26 @@ nothrow:
                  * GetFullPathName isn't it, but use it anyway.
                  */
                 // First find out how long the buffer has to be.
-                auto fullPathLength = GetFullPathNameW(&wname[0], 0, null, null);
+                const fullPathLength = GetFullPathNameW(&wname[0], 0, null, null);
                 if (!fullPathLength) return null;
                 auto fullPath = new wchar[fullPathLength];
 
                 // Actually get the full path name
                 const fullPathLengthNoTerminator = GetFullPathNameW(
-                    &wname[0], cast(uint)fullPath.length, &fullPath[0], null /*filePart*/);
+                    &wname[0], fullPathLength, &fullPath[0], null /*filePart*/);
                 // Unfortunately, when the buffer is large enough the return value is the number of characters
                 // _not_ counting the null terminator, so fullPathLengthNoTerminator should be smaller
                 assert(fullPathLength > fullPathLengthNoTerminator);
 
                 // Find out size of the converted string
+                enum codepage = CP_ACP;
```

Comment:
LDC will switch to `CP_UTF8` here, as it will use cmdline args in UTF-8. This code here assumes the narrow filename to be encoded in the system default Windows ANSI code page, probably because they mostly originate from the cmdline, where DMD uses the cmdline args passed to the narrow `main()` C entry point. I guess this breaks down as soon as DMD is fed with a UTF-8 response file containing some non-ASCII filenames etc.

Reply:
DMD doesn't care for the encoding of response files, so assumes CP_ACP there aswell.

Reply:
> LDC will switch to `CP_UTF8` here, as it will use cmdline args in UTF-8.

So maybe the codepage should be a global configuration variable thatLDC can modify?

Reply:
Okaaay, so source files must be ASCII or Unicode, but the expected response file encoding is ACP. ;) - It's at least an inconvenience when using DMD as a library on Windows.

Reply:
Yeah, it's quite a mess. I won't dare to import modules with non-ascii names in the module name when CP_ACP encoded import dirs are concatenated with UTF8 filenames ;-) Output is probably even worse...

Reply:
> So maybe the codepage should be a global configuration variable that LDC can modify?

Yes, I moved it as a module-level private enum to the header, to adapt it once.